db.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const mongoose = require('mongoose');
+
+const uri = 'mongodb://localhost/happy-mate';
+
+
+
+const setupAndConnect = () => {
+  mongoose.connect(uri, {useMongoClient: true});
```

Comment:
Is `mongoose.connect(...)` async or sync?

Reply:
`setupAndConnect(...)` should take a callback argument since `mongoose.connect(...)` is async.